### PR TITLE
Fix argument parsing

### PR DESCRIPTION
**Pull request by cvinson830**
Fixes issue: https://github.com/triton-inference-server/model_analyzer/issues/279
Feature: https://github.com/triton-inference-server/tensorflow_backend#--backend-configtensorflowversionint
Exercises this server code: https://github.com/triton-inference-server/server/blob/main/src/servers/main.cc#L1132-L1159

Problem: The MA code was not handling properly the case where TWO equals signs exist
(eg, `--backend-config=tensorflow,version=2`)
Before, the code accidentally split on both equals signs:
```
['tritonserver',  '--backend-config', 'tensorflow,version', '2']
```
The fix is to 
1) split on the argument list by spaces
2) Then for each of those arguments, split on the `first` equals sign
After the fix, the code splits on only the first equals sign:
```
['tritonserver', '--backend-config', 'tensorflow,version=2']
```
Notice how `version=2` is now correctly part of the `--backend_config` value

### CODE REVIEW

**Review by Tabrizian — CHANGES_REQUESTED** [review]
Thanks for root causing this. Is it possible to add a method to https://github.com/triton-inference-server/model_analyzer/blob/main/model_analyzer/triton/server/server_config.py named `to_args_list` and then it returns a list that can be added to `cmd`?

I think in the current fix we may still have issues if the argument/value has spaces in it. For example:
```
triton_server_flags:
  backend_config: "my parameter"
```

The cmd in this case will look like: ['tritonserver', '--backend-config', '"my', 'parameter"'] instead of ['tritonserver', '--backend-config', '"my parameter"'].

**Comment by cvinson830** [comment]
Regarding your comment above: there are two issues there.  Let's break them down.

1) Abstract functionality into function: Good idea. Done.
2) Handle embedded spaces within the value:  I agree that this is needed, but 
-- outside of the scope of this PR
-- It's not a "quick fix" since there are many ways to implement this (regex, csv, pyparsing, re to name four) 
-- it's always existed (it's not a result of the fix to this PR)
-- We need to make triton server can handle it.  

Let's decouple the second issue from this PR.  OK with you?

File: model_analyzer/triton/server/server_config.py [hunk 1]
```diff
@@ -123,6 +123,31 @@ def to_cli_string(self):
...
+            the tritonserver.
+
+            Example:
+            input cli_string: "--model-control-mode=explicit 
+                --backend-config=tensorflow,version=2"
+
+            output: ['--model-control-mode', 'explicit', 
+                '--backend-config', 'tensorflow,version=2']
+        """
+        list = []
```
Comment (Tabrizian) [inline model_analyzer/triton/server/server_config.py:145]: Could you please only rename this `list` variable to `args_list`? `list` is already defined and might create weird issues. No need to rerun the CI after the variable change. Thank you.

Comment (cvinson830) [reply model_analyzer/triton/server/server_config.py]: Done.

**Review by Tabrizian — APPROVED** [review]

**Pull request merged**